### PR TITLE
[refactor](nereids) Refactor stats collection framework

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

1. Significantly reduce the footprint of ResultRow and improve the performance of analyze by group insert partition stats
2. Remove table_statistics table and remove support for `with auto` grammer
3. Fix bug that literalexpr of columnstats would be lost when serialize and send to other FEs
4. Make health threshold configurable

I've test this PR  on ssb dataset which table definition could be found under tools directory, after this PR the IO request could reduce at least 80% on it.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.16 seconds
 stream load tsv:          559 seconds loaded 74807831229 Bytes, about 127 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17161887726 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowColumnStatsStmt.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ public ShowResultSet constructResultSet(List<Pair<String, ColumnStatistic>> colu
             row.add(String.valueOf(p.second.numNulls));
             row.add(String.valueOf(p.second.dataSize));
             row.add(String.valueOf(p.second.avgSizeByte));
-            row.add(String.valueOf(p.second.minExpr == null ? "N/A" : p.second.minExpr.toSql()));
+            row.add(String.valueOf(p.second.maxExpr == null ? "N/A" : p.second.minExpr.toSql()));
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowColumnStatsStmt.java:146]: Is this `maxExpr` a typo?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/HMSAnalysisTask.java [hunk 2]
```diff
@@ -117,39 +116,14 @@ public void doExecute() throws Exception {
      * Get table row count and insert the result to __internal_schema.table_statistics
      */
     private void getTableStats() throws Exception {
-        // Get table level information. An example sql for table stats:
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/HMSAnalysisTask.java:120]: Please keep this `if (isPartitionOnly) ` branch, this branch maybe useful for partition statistics.  I think just modify the else branch is ok for now.

File: fe/fe-core/src/main/java/org/apache/doris/statistics/TableStatisticsCacheLoader.java [hunk 3]
```diff
@@ -1,60 +0,0 @@
...
-        } catch (DdlException e) {
-            LOG.debug("Fail to get table line number from table_statistics table. "
-                    + "Will try to get from data source.", e);
-        }
-        // Get row count by call TableIf interface getRowCount
-        // when statistic table doesn't contain a record for this table.
-        try {
-            TableIf table = Env.getCurrentEnv().getCatalogMgr().getCatalog(key.catalogId)
-                    .getDbOrDdlException(key.dbId).getTableOrAnalysisException(key.tableId);
-            long rowCount = table.getRowCount();
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/TableStatisticsCacheLoader.java:51]: As discussed offline, we need to keep this logic somewhere for not analyzed external tables.

**Comment by Kikyou1997** [comment]
run buildall

**Review by englefly — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.41 seconds
 stream load tsv:          546 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162204318 Bytes

**Review by englefly — DISMISSED** [review]

**Review by starocean999 — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.41 seconds
 stream load tsv:          546 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17161921055 Bytes

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**